packages/core/src/destination-kit/index.ts

```diff
@@ -220,6 +223,14 @@ export interface TransactionContext {
   setTransaction: (key: string, value: string) => void
 }
 
+export interface StateContext {
+  // setState reads the `context` field from the request
+  getState(key: string, cb?: (res?: string) => any): any
+  // setState sets values in the `setContext` field in the response
+  // values set on the response will be available on subsequent requests
+  setState(key: string, value: string, ttl: { hour?: number; minute?: number; second?: number }): void
+}
```

Comment:
We should ensure the method names match the method names used in the monoservice. I think this will eliminate any confusion about whether these methods correspond to methods used in the monoservice. 

```suggestion
export interface StateContext {
  // getRequestContext reads the `context` field from the request
  getRequestContext(key: string, cb?: (res?: string) => any): any
  // setResponseContext sets values in the `setContext` field in the response
  // values set on the response will be available on subsequent requests
  setResponseContext(key: string, value: string, ttl: { hour?: number; minute?: number; second?: number }): void
}
```